File: dp/62.py
class Solution:
    def uniquePaths(self, m, n):
        """
        :type m: int
        :type n: int
        :rtype: int
        """
        matrix = [[1]*m for i in range(0,n)]
        print(matrix)
        
        for i in range(1, n):
            for j in range(1,m):
                matrix[i][j] = matrix[i-1][j] + matrix[i][j-1]
                
        return matrix[n-1][m-1]
        
        

File: dp/62.go
uniquePaths(m int, n int) int {
    var ret int
    dp := make([][]int, n)
    for i, _ := range dp {
        dp[i] = make([]int, m)
    }
    for i := 0; i < n; i++ {
        for j := 0; j < m; j++ {
            if i == 0 || j == 0 {
                dp[i][j] = 1
            }else {
                dp[i][j] = dp[i-1][j] + dp[i][j-1]
            }
            if ret < dp[i][j] {
                ret = dp[i][j]
            }            
        }
    }
    return ret
}
